### ISSUE TITLE
Issues with dynamic and on-submit validation

### ISSUE THREAD

**Issue by brandsis** [youngcut/kirby-form-block-suite#53]
Hi, Roman. I'm having problems with both dynamic  and on-submit validation. You can replicate my issues at [brandsistency.dev/brandsistency/contact](https://brandsistency.dev/brandsistency/contact).

To investigate, I have removed my custom form CSS and JS and have used your default styling. I do not have any custom block snippets. So, I think my form is as 'out of the box' as possible. I'm using the latest version 4.3.0.

<ins>**Dynamic validation**</ins>
Only seems to work if the relevant field is selected and entered first. In my form, the email field is the third field. If I click into it first, type an invalid email format (e.g. "abcdefg") and tab to the next field, the `invalid` attribute _is_ added to the `input` and the error message appears. However, if I start with another field, for example the first field (Name), fill the fields out in sequence tabbing as I go, when I get to the email field, fill that in with an invalid value and tab to the next field, the `invalid` attribute is _not_ added and the error message does _not_ appear.

(I'm not sure if there is dynamic validation of required fields but, if I click in a required field and click/tab away, there is no error message.)

<ins>**On-submit validation**</ins>
I'm getting the following scenarios:
- If I do not complete any fields and click the submit button, the `formblock__message--error` block appears, telling me which required fields are missing.
- If I complete the email field with an invalid value and immediately click submit, the field error message appears (as with dynamic validation above) but the `formblock__message--error` block does _not_ appear to say that other required fields are missing.
- If I complete any other fields, even just a non-required field, and click submit, there is no error message. I see a momentary flicker as the submit button changes to its 'in progress' text and then reverts to its 'normal' text.

Are these bugs or issues with my implementation?

**Comment by youngcut** [comment]
The problem is following:
<img width="534" alt="is" src="https://github.com/youngcut/kirby-form-block-suite/assets/4754476/aa69905c-87f0-4e11-bb48-c75c37cbd45e">
should be
<img width="516" alt="should_be" src="https://github.com/youngcut/kirby-form-block-suite/assets/4754476/fc7dfbf3-61a6-4fb7-be70-81909cfb9c3a">

Something insert an extra ' in the response.
What kind of plugin you have installed? Is there a compressing plugin activated?

**Comment by brandsis** [comment]
> The problem is following: <img alt="is" width="534" src="https://private-user-images.githubusercontent.com/4754476/341917780-aa69905c-87f0-4e11-bb48-c75c37cbd45e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyMjMyNDgsIm5iZiI6MTcxOTIyMjk0OCwicGF0aCI6Ii80NzU0NDc2LzM0MTkxNzc4MC1hYTY5OTA1Yy04N2YwLTRlMTEtYmI0OC1jNzVjMzdjYmQ0NWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjRUMDk1NTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDNiNDljYzU3YzhjYjRmYzBmOWRiMDk3ZWU5Y2I2YmU2ZWI5YWJlODExODdhOTQ4MjY3Zjg2NjEwNjZlNDliZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1qRkwqByiRSalWDGAK8I9czYgE0nLIzvun9kWwgoZz8"> should be <img alt="should_be" width="516" src="https://private-user-images.githubusercontent.com/4754476/341917784-fc7dfbf3-61a6-4fb7-be70-81909cfb9c3a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyMjMyNDgsIm5iZiI6MTcxOTIyMjk0OCwicGF0aCI6Ii80NzU0NDc2LzM0MTkxNzc4NC1mYzdkZmJmMy02MWE2LTRmYjctYmU3MC04MTkwOWNmYjljM2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjRUMDk1NTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzMyMDNkYzAyY2NiMTNlNGM3NDVhZTMwNjNjYTkxZGQzYzJhZTM0YjY4MmEyY2FmZDg2YTZiN2U4MjU0ZGZkYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.nJLGsES_VAzGjTf_qUogcc93QP85pqjRBO4GB7Ek1Fg">

Roman, sorry if this is a stupid question but in what view are you seeing `class='\"formblock__message--hidden\"'` etc.? If I look at Dev Tools or at the page source, I just see `class="formblock__message--hidden"` with no `\` and no `'`.

**Comment by youngcut** [comment]
On enter a value (and leaving the field) there is a request going on: 
<img width="676" alt="Bildschirmfoto 2024-06-25 um 13 45 33" src="https://github.com/youngcut/kirby-form-block-suite/assets/4754476/2acc4561-f7b4-470f-9e48-ef653de72385">
This `'\"` causes, that the browser insert `""` and the HTML is not more valid anymore.

It seems, your website works now. Can i close this issue?

**Comment by youngcut** [comment]
It's a known issue: https://github.com/youngcut/kirby-form-block-suite/issues/15

**Comment by brandsis** [comment]
> It seems, your website works now. Can i close this issue?

My [production website](https://brandsistency.com/contact) works because I have custom JS validating fields on focusout and on submission. Obviously, I'd prefer not to have to do that.

In my [test website](https://brandsistency.dev/brandsistency/contact), I removed the Minify HTML plugin and the issue does go away. Form Block Suite is, of course, more important than Minify HTML for functionality. To check any performance impact, I just tried GTMetrix and it doesn't seem that removing Minify HTML has hurt the site's performance too much.  I'll do some comparison of performance with and without Minify HTML but it looks like I might be able to live without it.

I see you've closed the issue. I'll reply again if I encounter any further related problems.

Thanks, Roman.
